toolchain/diagnostics/diagnostic_emitter.h

```diff
@@ -155,6 +181,55 @@ struct DiagnosticBase {
 template <typename LocationT>
 class DiagnosticEmitter {
  public:
+  class DiagnosticBuilder {
```

Comment:
Could you add some comments for the class and its public API?

Also, would `Build` be a reasonable name given that its a nested class?

Reply:
Wouldn't `Build` be a name conflict with the `Build` function?

I meant, I prefer a longer name for explicitness. But I think it's also helpful for unique names. And I don't think it matters _too_ much since users wouldn't right this.

Reply:
Sorry, I meant `Builder` instead of `DiagnsoticBuilder`.

Anyways, not worried about the name much, whatever works for you. I'm suggesting some additional information and detail in the comments that would help me out as a reader of the API and code.